src/unittest/Dasm.cc

```diff
@@ -0,0 +1,55 @@
+#include "Dasm.hh"
+#include "EmuTime.hh"
+#include "MSXCPUInterface.hh"
+
+#include "catch.hpp"
+
+#include <iostream>
+
+using namespace openmsx;
+
+TEST_CASE("instructionLength, dasm")
+{
+	std::array<uint8_t, 4> opcode = {0, 0, 0, 0};
+	std::string dest;
+
+	bool end = false;
+	while (!end) {
+		auto len = instructionLength(opcode).value_or(0);
+		assert(1 <= len);
+		assert(len <= 4);
+
+		dest.clear();
+		auto dasmLength = dasm(std::span(opcode.data(), len), 0, dest);
+
+		CHECKED_ELSE( len == dasmLength ) {
+			std::stringstream ss;
+			ss << std::hex;
+
+			for (unsigned j = 0; j < opcode.size(); ++j) {
+				ss << std::setw(2) << std::setfill('0') << int(opcode[j]) << ' ';
+			}
+			ss << '(' << dest << ")\n";
+			INFO( "opcode: " << ss.str());
+
+			auto instructionLength = len;
+			CHECK( dasmLength == instructionLength );
+		}
+
+		// goto next instruction
+		unsigned idx = len - 1;
+		while (true) {
+			if (opcode[idx] == 255) {
+				opcode[idx] = 0;
+				if (idx == 0) {
+					end = true;
+					break;
+				}
+				--idx;
+			} else {
+				++opcode[idx];
+				break;
+			}
+		}
+	}
```

Comment:
In my version of the test I also checked that we had exactly `2904196` iterations of this loop. I admit I did not calculate what this value should be. I only did some quick manual inspection of the output and then trusted the number that I got from running the test. But at least this will detect future (accidental?) changes.

I don't know if it's a good idea to add this check.